Add argument group

pavalos6401 opened the pull request:
Here is my first attempt at adding groups. 

This is to address issue #48 

If there is more to be done for this pull request, please let me know.

mivade commented on argparse_dataclass.py:
```diff
@@ -416,7 +416,32 @@ def _add_dataclass_options(
                     )
-        parser.add_argument(*args, **kwargs)
+
+        if "group" in field.metadata:
```
Can this be made into a method and unit tested?

mivade reviewed (commented):
Looks good to start with. In addition to some unit testing could you please add a usage example?

pavalos6401 commented:
Hey, I've moved the code to its own function and cleaned it up a bit. I also added documentation in the readme for some simple usage examples. I'll get started on the unit tests, but I might not get to it till tomorrow.

pavalos6401 commented:
I have added tests. Let me know if they make sense, or if you expected a different way to test them.

mivade commented:
Test failures appear to be related to formatting. If you run black against the code that should resolve.

pavalos6401 commented:
I've run black now, it seems it was one line that was too long.

mivade commented:
Thanks!
